js/services/contact_service.js

```diff
@@ -223,9 +223,6 @@ angular.module('contactsApp')
 		newContact.uid(newUid);
 		newContact.setUrl(addressBook, newUid);
 		newContact.addressBookId = addressBook.displayName;
-		if (_.isUndefined(newContact.fullName()) || newContact.fullName() === '') {
```

Comment:
We still need to check if the fullName of newContact is undefined. So please add the first half of this check back in.